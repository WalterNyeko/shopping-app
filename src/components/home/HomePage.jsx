import React, { Fragment } from 'react';
import UpperNav from '../../containers/commons/UpperNav';
import SideNav from '../../containers/commons/SideNav';

const HomePage = (props) => {
    const { content, departmentId } = props;
    return (
        <Fragment>
            <UpperNav/>
            <SideNav 
                content={content}
                departmentId={departmentId}/>
        </Fragment>
    )
}
export default HomePage;
