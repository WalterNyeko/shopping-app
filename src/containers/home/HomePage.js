import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import HomePageComponent from "../../components/home/HomePage";
import { getDepartments } from "../../store/actions/Departments";

class HomePage extends Component {
  componentWillMount = () => {
    const { getDepartments } = this.props;
    getDepartments();
  };

  render() {
    const {
      departmentId,
      allDepartments: { departments },
      handleSearch,
      handleInputChange,
      dontRender,
      content,
      categoryId
    } = this.props;
    return (
      <Fragment>
        <HomePageComponent
          content={content}
          dontRender={dontRender}
          categoryId={categoryId}
          departmentId={departmentId}
          departments={departments}
          handleSearch={handleSearch}
          handleInputChange={handleInputChange}
        />
      </Fragment>
    );
  }
}
const mapStateToProps = state => ({
  allDepartments: state.departments
});
export default connect(
  mapStateToProps,
  { getDepartments }
)(HomePage);
