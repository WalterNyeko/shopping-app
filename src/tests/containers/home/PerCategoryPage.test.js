import { shallow } from 'enzyme';
import React from 'react';
import PerCategory from '../../../containers/home/PerCategoryPage';
import PerCategoryComponent from '../../../components/home/PerCategoryPage';
import HomePage from '../../../containers/home/HomePage';

describe('PerCategory Page', () => {
    const props = {
        match: {
            params: {
                categoryId: 1
            }
        },
        allDepartments: {
            departments: []
        }
    }
 
    const wrapper = shallow(
        <PerCategory {...props}/>);
    const componentWrapper = shallow(
        <PerCategoryComponent {...props}/>);
    it('should render without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render component appropriately', () => {
        expect(componentWrapper.find(HomePage)).toHaveLength(1);
    });
})