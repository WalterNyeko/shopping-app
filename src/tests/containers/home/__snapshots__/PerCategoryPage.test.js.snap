// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerCategory Page should render without crashing 1`] = `ShallowWrapper {}`;
